src/table.rs

```diff
@@ -1,8 +1,46 @@
+//! Collecting parsed zoneinfo data lines into a set of time zone data.
+//!
+//! This module provides the `Table` struct, which is able to take parsed
+//! lines of input from the `line` module and coalesce them into a single
+//! set of data.
+//!
+//! It’s not as simple as it seems, because the zoneinfo data lines refer to
+//! each other through strings: lines of the form “link zone A to B” could be
+//! *parsed* successfully but still fail to be *interpreted* successfully if
+//! “B” doesn’t exist. So it has to check every step of the way—nothing wrong
+//! with this, it’s just a consequence of reading data from a text file.
+//!
+//! This module only deals with constructing a table from data: any analysis
+//! of the data is done elsewhere.
+//!
+//!
+//! ## Example
+//!
+//! ```
+//! use parse_zoneinfo::line::{Zone, Link, LineParser};
+//! use parse_zoneinfo::table::{TableBuilder};
+//!
+//! let parser = LineParser::new();
+//! let zone = parser.parse_zone("Zone  Pacific/Auckland  11:39:04  -  LMT  1868  Nov  2").unwrap();
+//! let link = parser.parse_link("Link  Pacific/Auckland  Antarctica/McMurdo").unwrap();
```

Comment:
If it is because of this doctest: I changed to to show how parse-zoneinfo is expected to be used in https://github.com/chronotope/parse-zoneinfo/pull/12/commits/536ac28a54f0b851a6355fada32b9df5bf3c89cd.

Reply:
It is because of that doctest. I want to keep this merge commit as small as possible -- the doctest is what it is, and then in future commits/PRs we can clean all this stuff up.